xarray/core/variable.py

```diff
@@ -442,7 +454,9 @@ def _broadcast_indexes_advanced(self, key):
                 variables.append(variable)
         variables = _broadcast_compat_variables(*variables)
         dims = variables[0].dims  # all variables have the same dims
-        key = tuple(variable.data for variable in variables)
+        # overwrite if there is integers
+        key = tuple(k if isinstance(k, integer_types) else variable.data
```

Comment:
Here I overwrite Variable by an integer if possible, resulting in much cleaner codes in `indexing._unbroadcast_indexes`.